homeassistant/components/sensor/paloalto.py

```diff
@@ -0,0 +1,266 @@
+"""
+Support for getting statistical data from a Palo Alto device.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.paloalto/
+"""
+import ssl
+import logging
+import urllib.request
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+import xml.etree.ElementTree as ET
+
+from enum import Enum
+from datetime import timedelta
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_NAME, CONF_API_KEY, CONF_IP_ADDRESS,
+                                 CONF_SSL, CONF_VERIFY_SSL,
+                                 CONF_MONITORED_CONDITIONS)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'PaloAlto'
+DEFAULT_SSL = False
+DEFAULT_VERIFY_SSL = True
+
+CONST_COMMAND = "COMMAND"
+CONST_OPS_ENDPOINT = '/api/?type=op&cmd=COMMAND'
+CONST_CONFIG_ENDPOINT = '/api/?type=config&action=get&xpath=COMMAND'
+
+PA_OPS_ACTIVE_USERS = "<show><admins></admins></show>"
+PA_CONF_SYS_INFO = "<show><system><info></info></system></show>"
+PA_CONF_GP_USERS = "<show><global-protect-portal><current-user>" \
+                    "</current-user></global-protect-portal></show>"
+PA_CONF_TEMPERATURE = "<show><system><environmentals><thermal>" \
+                        "</thermal></environmentals></system></show>"
+
+SCAN_INTERVAL = timedelta(seconds=120)
+
+MONITORED_CONDITIONS = {
+    'host_name': ['Host Name', 'x', 'mdi:fire'],
+    'up_time': ['Up Time', 'x', 'mdi:clock'],
```

Comment:
You are not allowed to have changing times in Home Assistant.

Reply:
Can you please provide more info on what you mean by `changing times` not allowed? I don't think I understood. This `monitored_conditions` section is no different from other similar sensors we already have - weather sensors, pi-hole...etc.